design-documents/graph-ql/coverage/deprecate-createEmptyCart-add-createGuestCart.md

```diff
@@ -0,0 +1,49 @@
+# Deprecate and replace `Mutation.createEmptyCart`
+
+## What
+
+- Mark `Mutation.createEmptyCart` as deprecated
+- Add `Mutation.createGuestCart` as a replacement that includes `Cart` in its return type
+
+## Why
+
+This PR started from a Slack discussion about mutations and client-side caching. At one point [@sirugh](https://github.com/sirugh) mentioned:
+
+> This makes me wonder if the createCart mutation should return a cart type instead of just the id
+
+There are 2 problems that I'm aware of with the current schema returning just an `ID` instead of the `Cart` type:
+
+1. Requires manual cache handling with libraries like Apollo, to link the `ID` back to a `Cart` object
+2. Requires 2 round trips for the client to get cart data for a guest. Even though the cart is empty, it still has defaults the pwa-studio components will use to render an empty cart
+
+During discussions, it was also noted that `createEmptyCart` has some behavior that's not clearly described by our schema. That is, If a customer is logged-in and has items in the cart, `createEmptyCart` returns an ID referencing a cart that is _not_ empty, which is extremely unintuitive. It was mentioned that the `pwa-studio` code [aliases this query](https://github.com/magento/pwa-studio/blob/38d652a4fbc797a4ac8ac0c3efa611003152c090/packages/venia-ui/lib/queries/createCart.graphql#L4) because of that.
```

Comment:
This sounds like a bug. Registered customers are supposed to use `customerCart` query to get cart object (either new or existing)